core-drawer-panel.html

```diff
@@ -188,157 +149,212 @@
          *
-         * @attribute forceNarrow
+         * @attribute rightDrawer
          * @type boolean
          * @default false
          */
-        forceNarrow: Boolean,
+        rightDrawer: {
+          type: Boolean,
+          value: false
+        },
 
         /**
-         * If true, swipe from the edge is disable.
+         * The panel that is being selected. `drawer` for the drawer panel and
+         * `main` for the main panel.
          *
-         * @attribute disableEdgeSwipe
-         * @type boolean
-         * @default false
+         * @attribute selected
+         * @type string
+         * @default null
          */
-        disableEdgeSwipe: Boolean
-      },
-
-      bind: {
-        forceNarrow: 'forceNarrowChanged'
-      },
-
-      listeners: {
-        click: 'onClick'
-      },
-
-      // TODO(ssorallen): Convert to 0.8 touch events
-      eventDelegates: {
-        trackstart: 'trackStart',
-        trackx: 'trackx',
-        trackend: 'trackEnd',
-        down: 'downHandler',
-        up: 'upHandler'
-      },
+        selected: {
+          reflect: true,
+          type: String,
+          value: null
+        },
 
-      configure: function() {
-        return {
-          defaultSelected: 'main',
-          disableEdgeSwipe: false,
+        swipeOverlayClass: {
+          computed: 'computeSwipeOverlayClass(narrow, disableEdgeSwipe)'
+        },
 
-          // Whether the user is dragging the drawer interactively.
-          dragging: false,
+        // The attribute on elements that should toggle the drawer on tap,
+        // also elements will automatically be hidden in wide layout.
+        drawerToggleAttribute: {
+          value: 'core-drawer-toggle'
+        },
 
-          drawerWidth: '256px',
+        // Whether the transition is enabled.
+        transition: false
 
-          // How many pixels on the side of the screen are sensitive to edge
-          // swipes and peek.
-          edgeSwipeSensitivity: 15,
+      },
 
-          forceNarrow: false,
+      computeCoreSelectorClass: function(narrow, transition, dragging, rightDrawer) {
+        return classNames({
+          dragging: dragging,
+          'narrow-layout': narrow,
+          'right-drawer': rightDrawer,
+          transition: transition
+        });
+      },
 
-          // Whether the browser has support for the transform CSS property.
-          hasTransform: ('transform' in this.style),
+      computeDrawerStyle: function(drawerWidth) {
+        return 'width:' + drawerWidth + ';';
+      },
 
-          // Whether the browser has support for the will-change CSS property.
-          hasWillChange: ('willChange' in this.style),
+      computeMainStyle: function(narrow, rightDrawer, drawerWidth) {
+        var style = '';
 
-          narrow: false,
+        style += 'left:' + ((narrow || rightDrawer) ? '0' : drawerWidth) + ';';
+        style += 'right:' + (rightDrawer ? (narrow ? '' : drawerWidth) : '') + ';';
```

Comment:
Nested ternaries are not my favorite thing in the world to read. Can we break some of this stuff apart?